src/components/Markets/index.tsx

```diff
@@ -123,52 +62,71 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
         });
 
         return <span className={className}>{cell}</span>;
-    }
+    }, []);
 
-    private mapRows = (cell: CellData) => {
+    const mapRows = React.useCallback((cell: CellData) => {
         const isChangeValue = typeof(cell) === 'string' && (cell.charAt(0) === '-' || cell.charAt(0) === '+');
 
-        return isChangeValue ? this.renderChange(cell as string) : cell;
-    };
+        return isChangeValue ? renderChange(cell as string) : cell;
+    }, [renderChange]);
 
-    private filterType = (headerKey: string, searchKey: string) => (item: CellData[]) => {
-        const typeIndex = this.props.headers ? this.props.headers.indexOf(headerKey) : this.defaultHeaders.indexOf(headerKey);
+    const filterType = React.useCallback((headerKey: string, key: string) => (item: CellData[]) => {
+        const typeIndex = (props.headers || DEFAULT_MARKET_HEADERS).indexOf(headerKey);
 
-        return (item[typeIndex] as string).includes(searchKey);
-    };
+        return (item[typeIndex] as string).includes(key);
+    }, [props.headers]);
 
-    private get filters() {
-        const { data } = this.props;
+    const createUniqueCurrencies = React.useCallback((currencies: string[], market: string) => {
+        const marketCurrencies = market.split('/').map((c: string) => c.trim());
+        const uniqueCurrencies = marketCurrencies.filter(c => !hasDuplicates(currencies, c));
+
+        return currencies.concat(uniqueCurrencies);
+    }, []);
```

Comment:
Need to rewrite this.
1. Use "base_unit" and "quote_unit" instead of splitting market name
2. currencies.concat(filtered_currencies) looks weird. Looks like it needs to be rewritten

Reply:
I'm not going to change logic which was written previously. This PR is about hooks

Reply:
this is old code, which I have not touched.

Reply:
I don't think this one is important to keep this PR from going ahead. Again, this PR is about moving to hooks some of components, not some logic cases written before.